fix: refactor primary chain

janniks opened the pull request:
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5610259162).<!-- Sticky Header Marker -->

- refactors for validating subnet and working with the correct chain id in different code paths

### Explanation
I added a "primary" chain id (which is the parent network, if one is available).
The primary chainId/network is the one used for address detection, transaction version etc. (mainly the checks for mainnet/testnet).
The actual chainId is used mainly for signing (and thus is on the network object constructed from the user-provided url.

I hope this makes sense. It will likely not hold up once nested subnets emerge (which isn't too soon I hope)

janniks commented:
That sounds better 😅 Not sure how I landed on this. I can rename chainId to subnetChainId and parentChainId back to the original chainId

kyranjamie reviewed (commented):
Agree with Fara on primary/parent naming mismatch, otherwise this works. Are other changes needed?

janniks commented:
Yep, great point. This evolved with my understanding of the different IDs.

Rewrote to use subnetChainId -- works a lot cleaner. I had an additional incorrect assumption, so a bunch of other stuff changed. Added comments to explain.

lmk if the add-network function is getting too big and you want more thoughtful refactors.
lmk if I should rebase to a readable history, or if you'll merge/squash anyway.

fbwoolf reviewed (approved):
Thanks for making the changes @janniks, this makes sense to me now. 👍 Looks good.

markmhx commented:
@janniks Do you want to make more changes here before merging or is this ready to release?

janniks commented:
I think we're good to go on this one. Can be merged ✌🏻
